### DISCUSSION TITLE
Grid expressed in Easting Northing

### DISCUSSION THREAD

**Discussion by marstonsward** [Turbo87/utm#96] [General]
I'm wondering if I can define a rectilinear grid of dims (5000, 8000) and range 90 : -90 and 180 : -180.
Noticed that I had to limit the lats between 80 : -80. Is that always true? Then I noticed that utm is requiring a square grid, i.e., that the lat lon are of the same size. 
Is there a way to create a grid defined in E/N from one already defined in lat lon?

**Comment by bartvanandel** [comment]
I don't understand exactly what you're trying to do, but when you're talking about a "grid" of 5000 x 8000, you'd need to actually create 5000 x 8000 = 40,000,000 lat/lon pairs. Latitude of each point goes into one list, longitude in another, which is why those lists need to be the same size.

The maximum latitude, according to official UTM specifications, see e.g. [Wikipedia](https://en.m.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system#Locating_a_position_using_UTM_coordinates "Universal Transverse Mercator coordinate system - Locating a position using UTM coordinates"), is 84 degrees North and 80 degrees South. This is also what `utm` checks: it deems values between -80 and 84 valid. You can find this bounds check in the `from_latlon()` function.
